### ISSUE TITLE
Make client timeouts configurable

### ISSUE THREAD

**Issue by GeorgFleig** [TwinProduction/gatus#58]
Hi,

first of all thank you for providing this great tool!

I am currently testing it in our environment and noticed that some health endpoints I like to monitor do not respond within 10 seconds as they check multiple system dependencies themselves. It would be great to be able to configure the timeout for HTTP (and TCP) requests.

Furthermore, I noticed that the user-agent is currently `Go-http-client/2.0` which makes it hard to identify the requests as the ones coming from Gatus. Would be great if this became a config option as well or a string that contains the name of the software.

**Comment by TwinProduction** [comment]
Great idea!

Something like
```yaml
services:
  - name: example
    url: "https://example.org/"
    interval: 30s
    client:
      timeout: 30s
    conditions:
      - "[STATUS] == 200"
```
Would work for setting timeouts, but I have a feeling configuring the user-agent at the service level may be a bit too granular, although I could see a use for it if somebody wanted to test user-agent specific behaviors.

For now, a hardcoded custom user agent like `Gatus/1.0` could do the trick to at least make Gatus requests identifiable.

**Comment by GeorgFleig** [comment]
I wasn't thinking of having these options on a service level in the first place. I'd be fine with a global config option. But you're right, it makes sense for the timeout and there might even be scenarios for the user-agent 👍

**Comment by TwinProduction** [comment]
@GeorgFleig I've added the "User-Agent: Gatus/1.0" header to requests that don't specify a custom User-Agent header.
As for services that require a different User-Agent header, it can already be specified in the configuration:

```yaml
services:
  - name: example
    url: "https://example.org/"
    headers:
      User-Agent: "Mozilla/5.0"
    conditions:
      - "[STATUS] == 200"
```

**Comment by TwinProduction** [comment]
Released the default custom user agent in [v1.11.0](https://github.com/TwinProduction/gatus/releases/tag/v1.11.0)

**Comment by TwinProduction** [comment]
Timeout is now configurable through [client configuration](https://github.com/TwinProduction/gatus#client-configuration) -- forgot to close this.
